app/kumactl/cmd/install/install_dns.go

```diff
@@ -92,3 +101,48 @@ func getCoreFileForwardVerb(corefile string) string {
 
 	return "forward"
 }
+
+func handleCoreDNS(clientset *kubernetes.Clientset, cpaddress string) (*v1.ConfigMap, error) {
+	corednsConfigMap, err := clientset.CoreV1().ConfigMaps("kube-system").Get(context.TODO(),
+		"coredns", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.Contains(corednsConfigMap.Data["Corefile"], "mesh:53") {
+		forwardVerb := getCoreFileForwardVerb(corednsConfigMap.Data["Corefile"])
+		toappend := fmt.Sprintf(corednsAppendTemplate, forwardVerb, cpaddress)
+		corednsConfigMap.Data["Corefile"] += toappend
+	}
+	return corednsConfigMap, nil
+}
+
+func handleKubeDNS(clientset *kubernetes.Clientset, cpaddress string) (*v1.ConfigMap, error) {
+	corednsConfigMap, err := clientset.CoreV1().ConfigMaps("kube-system").Get(context.TODO(),
+		"kube-dns", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.Contains(corednsConfigMap.Data["stubDomains"], "\"mesh\"") {
+		toappend := fmt.Sprintf(kubednsAppendTemplate, cpaddress)
+		corednsConfigMap.Data["stubDomains"] += toappend
```

Comment:
you cannot just append it. It's an object and it may contain other stuff.
For example if you have:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: kube-dns
  namespace: kube-system
data:
  stubDomains: |
    {"global": ["192.168.0.1"]}
```

Kuma should modify it to
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: kube-dns
  namespace: kube-system
data:
  stubDomains: |
    {
      "global": ["192.168.0.1"],
      "mesh": ["192.168.0.2:port"]
    }
```